internal/core/src/exec/expression/UnaryExpr.cpp

```diff
@@ -196,6 +262,106 @@ PhyUnaryRangeFilterExpr::ExecRangeVisitorImplArray() {
     return res_vec;
 }
 
+template <typename T>
+VectorPtr
+PhyUnaryRangeFilterExpr::ExecArrayEqualForIndex(bool reverse) {
+    typedef std::
+        conditional_t<std::is_same_v<T, std::string_view>, std::string, T>
+            IndexInnerType;
+    using Index = index::ScalarIndex<IndexInnerType>;
+    auto real_batch_size = GetNextBatchSize();
+    if (real_batch_size == 0) {
+        return nullptr;
+    }
+
+    // get all elements.
+    auto val = GetValueFromProto<proto::plan::Array>(expr_->val_);
+    if (val.array_size() == 0) {
+        // rollback to bruteforce. no candidates will be filtered out via index.
+        return ExecRangeVisitorImplArray<proto::plan::Array>();
+    }
+
+    // cache the result to suit the framework.
+    auto batch_res =
+        ProcessIndexChunks<IndexInnerType>([this, &val, reverse](Index* _) {
+            std::unordered_set<IndexInnerType> elements;
+            boost::container::vector<IndexInnerType> elems;
+            for (auto const& element : val.array()) {
+                auto e = GetValueFromProto<IndexInnerType>(element);
+                if (elements.find(e) == elements.end()) {
+                    elems.push_back(e);
+                }
+                elements.insert(e);
+            }
```

Comment:
no need to use element set, just std::find(elems.begin(), elems.end(), target) ! = elems.end()

Reply:
OK.